Tests/FuzzilliTests/LifterTest.swift

```diff
@@ -363,6 +363,31 @@ class LifterTests: XCTestCase {
 
         XCTAssertEqual(lifted_program,expected_program)
     }
+
+    func testHoleyArrayLifting(){
+        let fuzzer = makeMockFuzzer()
+        let b = fuzzer.makeBuilder()
+
+        var initialValues = [Variable]()
+        initialValues.append(b.loadInt(1))
+        initialValues.append(b.loadInt(2))
+        initialValues.append(b.loadUndefined())
+        initialValues.append(b.loadInt(4))
+        initialValues.append(b.loadUndefined())
+        initialValues.append(b.loadInt(5))
+        b.createArray(with: initialValues)
```

Comment:
How about adding the following code here as well:
```
let v = b.loadString("foobar")
b.reassign(v, b.loadUndefined())
initialValues.append(v)
```
and then checking that the expected program is (I think)
```
let v6 = 42;
v6 = undefined;
const v8 = [1,2,,4,,6,v6];
```
?

Just to be sure that the lifter really only treats the `undefined` literal as hole and not other variables that happen to contain the undefined value.

Reply:
good spot! I've updated the testcase to include the inlining test